src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/ChocolateFactorySolver.java

```diff
@@ -0,0 +1,165 @@
+	}
+
+	@Override
+	protected List<ColorHighlight> getColors(String[] groups, Int2ObjectMap<ItemStack> slots) {
+		final Int2ObjectMap<Rabbit> cpsIncreaseFactors = new Int2ObjectLinkedOpenHashMap<>(6);
+		for (int i = 29; i <= 33; i++) { // The 5 rabbits slots are in 29, 30, 31, 32 and 33.
+			ItemStack item = slots.get(i);
+			if (item.getItem() != Items.PLAYER_HEAD || item.isEmpty()) continue;
+
+			int finalI = i; //Java, pfft.
+			getRabbit(item).ifPresent(rabbit -> cpsIncreaseFactors.put(finalI, rabbit));
+		}
+
+		//Coach is in slot 42 while the factory info item is in slot 45.
+		getCoach(slots.get(45), slots.get(42)).ifPresent(coach -> cpsIncreaseFactors.put(42, coach));
+		if (cpsIncreaseFactors.isEmpty()) return List.of(); //Something went wrong.
+
+		OptionalLong totalChocolate = getTotalChocolate(slots.get(13));
+
+		List<Int2ObjectMap.Entry<Rabbit>> sorted = cpsIncreaseFactors.int2ObjectEntrySet()
+		                                                             .stream() //Compare cost/cpsIncrease rather than cpsIncrease/cost to avoid getting close to 0 and losing precision.
+		                                                             .sorted(Comparator.comparingDouble(entry -> entry.getValue().cost() / entry.getValue().cpsIncrease())) //Ascending order, lower = better
+		                                                             .dropWhile(entry -> entry.getValue().cost == 0)
+		                                                             .collect(Collectors.toCollection(LinkedList::new));
+
+		Int2ObjectMap.Entry<Rabbit> bestEntry = sorted.removeFirst();
+		if (totalChocolate.isEmpty() || bestEntry.getValue().cost < totalChocolate.getAsLong()) return List.of(ColorHighlight.green(bestEntry.getIntKey()));
+
+		for (Int2ObjectMap.Entry<Rabbit> entry : sorted) {
+			if (entry.getValue().cost > totalChocolate.getAsLong()) continue;
+			return List.of(ColorHighlight.green(entry.getIntKey()), ColorHighlight.yellow(bestEntry.getIntKey()));
+		}
+		return List.of(ColorHighlight.green(bestEntry.getIntKey()));
+	}
+
+	/**
+	 * Utility method.
+	 */
+	private String getConcattedLore(ItemStack item) {
+		return concatenateLore(ItemUtils.getLore(item));
+	}
+
+	/**
+	 * Concatenates the lore of an item into one string.
+	 * This is useful in case some pattern we're looking for is split into multiple lines, which would make it harder to regex.
+	 */
+	private String concatenateLore(List<Text> lore) {
+		StringBuilder stringBuilder = new StringBuilder();
+		for (int i = 0; i < lore.size(); i++) {
+			stringBuilder.append(lore.get(i).getString());
+			if (i != lore.size() - 1) stringBuilder.append(" ");
+		}
+		return stringBuilder.toString();
+	}
+
+	private Optional<Rabbit> getCoach(ItemStack cpsItem, ItemStack coachItem) {
+		if (coachItem.isEmpty() || cpsItem.isEmpty() || coachItem.getItem() != Items.PLAYER_HEAD || cpsItem.getItem() != Items.COCOA_BEANS) return Optional.empty();
```

Comment:
```suggestion
		if (!coachItem.isOf(Items.PLAYER_HEAD) || !cpsItem.isOf(Items.COCOA_BEANS)) return Optional.empty();
```
same as before, use of isOf and removal of isEmpty